hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/utils/Pipelines.java

```diff
@@ -482,7 +486,19 @@ public static String opName(String operatorN, Configuration conf) {
   }
 
   public static String opUID(String operatorN, Configuration conf) {
-    return "uid_" + operatorN + "_" + getTablePath(conf);
+    String uid = "uid_" + operatorN + "_" + getTablePath(conf);
+    String finalUid = uid;
+    synchronized (Pipelines.class) {
+      ArrayList<String> operators = operatorUidMap.get(uid);
+      if (operators != null) {
+        finalUid = uid + "_" + operators.size();
```

Comment:
Just bookkeep the count should be fine ? And the key could be operatorN

Reply:
Adjust operator-uid-map to operator-counter.